src/utils/EventUtils.js

```diff
@@ -50,3 +51,35 @@ export function canEditContent(mxEvent) {
         mxEvent.getOriginalContent().msgtype === "m.text" &&
         mxEvent.getSender() === MatrixClientPeg.get().getUserId();
 }
+
+export function findPreviousEditableEvent(room, fromEventId = undefined) {
+    const liveTimeline = room.getLiveTimeline();
+    const events = liveTimeline.getEvents();
+    let startFromIdx = events.length - 1;
+    if (fromEventId) {
+        const fromEventIdx = findLastIndex(events, e => e.getId() === fromEventId);
+        if (fromEventIdx !== -1) {
+            startFromIdx = fromEventIdx - 1;
+        }
+    }
+    const nextEventIdx = findLastIndex(events, e => !shouldHideEvent(e) && canEditContent(e), startFromIdx);
```

Comment:
This and the other function need an early stop condition. It would suck to hit the up arrow in a room which you idle in and have riot freeze for a while.

Reply:
I *guess* it wouldn't be common to want to quickly edit an event more than 100 events up in the timeline, so I could limit the loop to something like that.

Reply:
Implemented a limit of 100. Wdyt?

Reply:
+1000 thank you